### PR TITLE
Add method to unset actor request ref tracker

### PR DESCRIPTION
**Pull request by MellowYarker**

### CODE REVIEW

File: src/workerd/io/request-tracker.c++ [hunk 1]
```diff
@@ -28,6 +28,7 @@ RequestTracker::ActiveRequest::~ActiveRequest() {
       }
     }
   }
+  parent = nullptr;
```
Comment (bcaimano) [inline src/workerd/io/request-tracker.c++:31]: nit: You don't have to worry about this, the contents of the maybe will be dtor'd as part of destroying the `ActiveRequest`.

File: src/workerd/io/worker.h [hunk 2]
```diff
@@ -736,6 +736,11 @@ class Worker::Actor final: public kj::Refcounted {
   // Runs an alarm in this actor. This function will handle deduplicating multiple alarms. If this
   // isn't a duplicate alarm, func will be called to run the alarm.
 
+  void unsetRequestTracker() {
```
Comment (bcaimano) [inline src/workerd/io/worker.h:739]: I think there is a `TODO(soon)` below that we probably want to remove since we're relying on the maybeness.

**Review by bcaimano — APPROVED** [review]
Makes sense to me!

**Comment by MellowYarker** [comment]
Thanks for the review, Ben!

**Review by jqmmes — APPROVED** [review]

**Pull request merged**